packages/core/src/routes/experience/middleware/koa-interaction-session.ts

```diff
@@ -0,0 +1,27 @@
+import type { MiddlewareType } from 'koa';
+
+import { type WithLogContext } from '#src/middleware/koa-audit-log.js';
+import type TenantContext from '#src/tenants/TenantContext.js';
+
+import InteractionSession from '../classes/interaction-session.js';
+
+export type WithInteractionSessionContext<ContextT extends WithLogContext = WithLogContext> =
+  ContextT & {
+    interactionSession: InteractionSession;
+  };
+
+/**
+ * @overview This middleware initializes the InteractionSession for the current request.
+ * The InteractionSession instance is used to manage all the data related to the current interaction.
+ * All the interaction data is stored using the oidc-provider's interaction session
+ * @see {@link https://github.com/panva/node-oidc-provider/blob/main/docs/README.md#user-flows}
```

Comment:
remember to update this part accordingly if you're changing the name of this context